src/arrayMethodJoin.js

```diff
@@ -4,8 +4,53 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let sep = separator;
+    let joinedText = '';
+
+    if (separator === null) {
+      sep = 'null';
+    }
+
+    if (this.length === 1) {
+      joinedText += this[0];
+
+      return joinedText;
+    } else if (this.length > 1) {
+      switch (this[0]) {
+        case null:
+          joinedText += '';
+          break;
+
+        default:
+          joinedText += this[0];
+          break;
+      }
+    }
+
+    if (this.length > 1) {
+      for (let i = 1; i < this.length; i++) {
+        let elem = this[i];
+
+        switch (this[i]) {
+          case null:
+            elem = '';
```

Comment:
Checklist p5

Reply:
if - statement( this.length > 1) was deleted